src/monitor/node_active_protocol.c

```diff
@@ -1060,6 +1065,47 @@ RemoveNode(AutoFailoverNode *currentNode)
 		currentNode->formationId,
 		currentNode->groupId);
 
+	/*
+	 * Adjust number-sync-standbys if necessary.
+	 *
+	 * otherNodesGroupList is the list of all the remaining nodes, and that
+	 * includes the current primary, which might be setup with replication
+	 * quorum set to true (and probably is).
+	 */
+	countSyncStandbys = CountSyncStandbys(otherNodesGroupList) - 1;
+
+	LogAndNotifyMessage(message, BUFSIZE,
+						"CountSyncStandbys: %d", countSyncStandbys);
+
+	if (countSyncStandbys < (formation->number_sync_standbys + 1))
```

Comment:
Is otherwise ever possible? I mean `countSyncStandbys  >= formation->number_sync_standbys + 1`?

We normally prevent it, still being defensive here is good, just want to ask in case I oversee some cases.

Reply:
That's the code that runs when we just dropped a standby. So it might be that we had two standby nodes and number_sync_standbys set to 1, all good. One standby removed. One standby left, number_sync_standbys now must be set to zero. That's what this test is all about.

Now, it could be that we had three standby nodes, and still had number_sync_standbys set to 1. Now a standby is removed. Two standby left, number_sync_standbys = 1, it's all good. That's your “otherwise” example, right?